pkg/controller/rmdworkload/rmdworkload_controller.go

```diff
@@ -147,46 +135,36 @@ func (r *ReconcileRmdWorkload) Reconcile(request reconcile.Request) (reconcile.R
 	// Nodestates and RMD pods are created with the same namespace by node_controller.
 	// This is the same namespace as the parent node, or default if not set.
 	rmdWorkloadName := rmdWorkload.GetObjectMeta().GetName()
-	nodeNamespaces := make(map[string]string)
-	for _, rmdNodeState := range rmdNodeStates.Items {
-		nodeNamespaces[rmdNodeState.Spec.Node] = rmdNodeState.GetObjectMeta().GetNamespace()
-	}
-	// Loop through nodes listed in RmdWorkload Spec, add/update workloads where necessary.
-	for _, nodeName := range rmdWorkload.Spec.Nodes {
-		// Get node service address
-		address, err := r.getPodAddress(nodeName, nodeNamespaces[nodeName])
-		if err != nil {
-			reqLogger.Error(err, "Failed to get pod address")
-			return reconcile.Result{}, err
-		}
 
-		activeWorkloads, err := r.rmdClient.GetWorkloads(address)
-		if err != nil {
-			reqLogger.Info("Could not GET workloads.", "Error:", err)
-			return reconcile.Result{}, err
-		}
+	wlsToChange, err := r.findWorkloadsToChange(request, rmdNodeStates, rmdWorkload)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
 
-		workload := rmd.FindWorkloadByName(activeWorkloads, rmdWorkloadName)
+	i := 0
```

Comment:
Using an iterator here is unreliable.

I originally thought that we could use a map of address to workload for Add and Update, similar to how it was done for DeleteWorkload. But I see now we also need the Node name parameter for that.

So I would suggest creating a new type:
````
type targetedNdoeInfo {
    nodeName      string
    rmdAddress     string
    workloadExists bool
}
```` 
and returning `[]targetedNodeInfo` from `findWorkloadsToChange`. Now inside `findWorkloadsToChange`, we can perform the check on `workload.UID` that currently exists at the higher level Reconcile, but instead use this check to update `targetedNodeInfo.workloadExists` for each iteration.

Then at the higher level `Reconcile` function, we loop over `[]targetedNodeInfo` and Add or Update based on each iteration's `workloadExists` field.